net-utils/src/lib.rs

```diff
@@ -439,6 +439,11 @@ pub fn bind_in_range(ip_addr: IpAddr, range: PortRange) -> io::Result<(u16, UdpS
     ))
 }
 
+pub fn bind_client_socket() -> io::Result<UdpSocket> {
+    bind_in_range(IpAddr::V4(Ipv4Addr::new(0, 0, 0, 0)), VALIDATOR_PORT_RANGE)
```

Comment:
Do you know why we limit to `VALIDATOR_PORT_RANGE`? (I know that predates this particular PR)

But regardless, I think calling this function `bind_in_validator_port_range()` makes the weirdness a little more clear.

Reply:
> Do you know why we limit to VALIDATOR_PORT_RANGE? (I know that predates this particular PR)

Sorry, I don't have a firm answer to that.  The only thing I can see is that the socket must be bound to some local address + port, so we just use any potential port that's available to the validator.

> But regardless, I think calling this function bind_to_validator_port_range() makes the weirdness a little more clear.

Will do

> Can we also make the caller pass in ip_addr: IpAddr rather than assuming 0.0.0.0

Definitely

Reply:
> > Do you know why we limit to VALIDATOR_PORT_RANGE? (I know that predates this particular PR)
>
> Sorry, I don't have a firm answer to that. The only thing I can see is that the socket must be bound to some local address + port, so we just use any potential port that's available to the validator.

We do this for server ports for the ease of setting up inbound firewall rules. 

Client ports typically don't require this though, but if so then this range ought to be configurable on the `solana-validator` command-line.  But yes, not directly relevant to this current PR :)